sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosVectorEmbedding.java

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.models;
+
+import com.azure.cosmos.implementation.Constants;
+import com.azure.cosmos.implementation.JsonSerializable;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * Embedding settings within {@link VectorEmbeddingPolicy}
+ */
+public final class CosmosVectorEmbedding {
+    @JsonProperty(Constants.Properties.PATH)
+    private String path;
+    @JsonProperty(Constants.Properties.VECTOR_DATA_TYPE)
+    private String cosmosVectorDataType;
+    @JsonProperty(Constants.Properties.VECTOR_DIMENSIONS)
+    private Long dimensions;
+    @JsonProperty(Constants.Properties.DISTANCE_FUNCTION)
+    private String cosmosVectorDistanceFunction;
+    private JsonSerializable jsonSerializable;
+
+    /**
+     * Constructor
+     *
+     * @param path path for the cosmosVectorEmbedding
+     * @param vectorDataType data type for the embedding
+     * @param dimensions dimensions for the embedding
+     * @param distanceFunction distanceFunction for the embedding
+     */
+    public CosmosVectorEmbedding(String path, String vectorDataType, Long dimensions, String distanceFunction) {
+        this.path = path;
+        this.cosmosVectorDataType = vectorDataType;
+        this.dimensions = dimensions;
+        this.cosmosVectorDistanceFunction = distanceFunction;
+    }
```

Comment:
Can this constructor be deleted? We have the no-args ctor and setter methods.

Reply:
Done